transfer-products/src/transfer_products.py

```diff
@@ -0,0 +1,175 @@
+import argparse
+import os
+from dataclasses import dataclass
+
+import boto3
+import boto3.s3.transfer
+import botocore.exceptions
+import hyp3_sdk
+import requests
+from dotenv import load_dotenv
+
+S3 = boto3.resource('s3')
+
+TARGET_BUCKET = 'hyp3-nasa-disasters'
+
+# TODO decide on appropriate extensions
+EXTENSIONS = ['_VV.tif', '_VH.tif', '_rgb.tif', '_dem.tif', '_WM.tif', '.README.md.txt']
+
+
+@dataclass(frozen=True)
+class ObjectToCopy:
+    source_bucket: str
+    source_key: str
+    target_key: str
+
+    # TODO if we migrate to an enterprise test deployment: remove this field
+    url: str
+
+
+class MissingEnvVar(Exception):
+    pass
+
+
+def get_existing_objects(target_prefix: str) -> frozenset[str]:
+    return frozenset(obj.key for obj in S3.Bucket(TARGET_BUCKET).objects.filter(Prefix=f'{target_prefix}/'))
+
+
+def get_objects_to_copy(
+        jobs: hyp3_sdk.Batch,
+        existing_objects: frozenset[str],
+        target_prefix: str,
+        extensions: list[str]) -> list[ObjectToCopy]:
+
+    objects_to_copy = []
+    for job in jobs:
+        assert job.succeeded()
+        source_bucket: str = job.files[0]['s3']['bucket']
+
+        zip_key: str = job.files[0]['s3']['key']
+        assert zip_key.endswith('.zip')
+
+        zip_url: str = job.files[0]['url']
+        assert zip_url.endswith('.zip')
+
+        for ext in extensions:
+            source_key = get_source_key(zip_key, ext)
+            target_key = get_target_key(source_key, job.name, job.job_id, target_prefix)
+
+            url = get_source_key(zip_url, ext)
+            assert url.endswith(source_key)
+
+            if target_key not in existing_objects:
+                objects_to_copy.append(
+                    ObjectToCopy(source_bucket=source_bucket, source_key=source_key, target_key=target_key, url=url)
+                )
+
+    return objects_to_copy
+
+
+def get_source_key(zip_key: str, ext: str) -> str:
+    return zip_key.removesuffix('.zip') + ext
+
+
+def get_target_key(source_key: str, job_name: str, job_id: str, target_prefix: str) -> str:
+    source_prefix, source_basename = source_key.split('/')
```

Comment:
this assumes there's only ever one `/` in the source key, which is true at the moment, but it's likely worth breaking off just the final token (although you'd also have to change the `assert`)